lib/matplotlib/widgets.py

```diff
@@ -613,43 +607,94 @@ def _clicked(self, event):
         """
         Directly (de)activate a check button by index.
 
-        *index* is an index into the original label list
-            that this object was constructed with.
-            Raises ValueError if *index* is invalid.
+        Default behaviour is to toggle the state, which can be controlled with
+        state=bool flag.
 
         Callbacks will be triggered if :attr:`eventson` is True.
 
+        Parameters
+        ----------
+        index : int
+            index into the original label list
+
+        state : bool
+            boolean value to set the target state of the button, regardless of
+            its current state. By default (no value specified),  the button
+            state gets toggled (checked or not). Only True or False values
+            are allowed for *state*
+
+        Raises
+        ------
+        ValueError
+            If *index* is invalid.
+
+        TypeError
+            If state is not boolean.
         """
-        if 0 > index >= len(self.labels):
+        if index < 0 or index >= len(self.labels):
             raise ValueError("Invalid CheckButton index: %d" % index)
 
         l1, l2 = self.lines[index]
-        l1.set_visible(not l1.get_visible())
-        l2.set_visible(not l2.get_visible())
+
+        if state is not _TOGGLE:
+            if not isinstance(state, bool):
+                raise TypeError('state can only be either True or False')
+            target_vis = state
+        else:
+            target_vis = not l1.get_visible()
+
+        l1.set_visible(target_vis)
+        l2.set_visible(target_vis)
 
         if self.drawon:
             self.ax.figure.canvas.draw()
 
+        self._exec_callbacks(self.labels[index].get_text())
+
+    def clear(self):
+        """Clear all checkboxes."""
+
+        for l1, l2 in self.lines:
+            l1.set_visible(False)
+            l2.set_visible(False)
+
+        if self.drawon:
+            self.ax.figure.canvas.draw()
+
+        # calling it with no label, as all checkboxes are being cleared
+        self._exec_callbacks(None)
+
+    def _exec_callbacks(self, current_label=None):
+        """Run all registered callbacks."""
+
         if not self.eventson:
             return
+
         for cid, func in self.observers.items():
-            func(self.labels[index].get_text())
+            func(current_label)
 
     def get_status(self):
         """
         returns a tuple of the status (True/False) of all of the check buttons
         """
         return [l1.get_visible() for (l1, l2) in self.lines]
 
+    def get_checked_labels(self):
+        """Returns a list of labels currently checked by user."""
+
+        return [l.get_text() for l, box_checked in
+                zip(self.labels, self.get_status())
+                if box_checked]
+
     def on_clicked(self, func):
         """
-        Connect the callback function *func* to button click events.
+        When the button is clicked, call *func* with button label
```

Comment:
I find the original description more to the point. This method connects a callback function. What you are describing is the execution of the callback. If you feel the need that this should be explained in the docstring, that should go in a second paragraph after "Connect the callback ..." (optional to me).